core/src/services/routing.js

```diff
@@ -662,18 +665,21 @@ class RoutingClass {
       } else {
         url.search = `?${RoutingHelpers.encodeParams(params)}`;
       }
-      history.replaceState(window.state, '', url.href);
+      if (!sessionStorage.getItem('historyState')) {
```

Comment:
Is **!sessionStorage.getItem('historyState')** check necessary here ? 🤔 
We check for **(sessionStorage.getItem('historyState'))** in line 643 above. Seems like in this line we are sure that history state does not exist already, otherwise code execution wouldn't arrive here at all.

Reply:
👍